Compile warning from `c-index-test.c` under Ubuntu 26.04 LTS

Whilst compiling clang on Ubuntu 26.04, I get this warning from c-index-test.c:
```
  /home/jontha01/llvm-project/clang/tools/c-index-test/c-index-test.c:298:23: warning: '_Generic' is a C11 extension [-Wc11-extensions]
    298 |     const char *sep = strchr(arg_string, ',');
        |                       ^
  /usr/include/string.h:265:3: note: expanded from macro 'strchr'
    265 |   __glibc_const_generic (S, const char *, strchr (S, C))
        |   ^
  /usr/include/aarch64-linux-gnu/sys/cdefs.h:838:3: note: expanded from macro '__glibc_const_generic'
    838 |   _Generic (0 ? (PTR) : (void *) 1,                     \
        |   ^
```

It appears that Ubuntu 26.04 LTS updates glibc's string.h so `strchr()`, `strrchr()` and `strstr()` are routed through _Generic-based macros. When Clang compiles `c-index-test.c` with `-std=gnu89` and `-pedantic`, that now triggers `-Wc11-extensions` warnings even though the source itself is not using C11 features.

I believe that `c-index-test.c` is compiled with `-std=gnu89` on purpose to act as a guard rail.

--- a/clang/tools/c-index-test/CMakeLists.txt
+++ b/clang/tools/c-index-test/CMakeLists.txt
@@ -12,6 +12,12 @@ if(NOT MSVC)
     SOURCE c-index-test.c
     PROPERTY COMPILE_FLAGS "-std=gnu89"
     )
+  if(CMAKE_C_COMPILER_ID MATCHES "Clang")
+    set_property(
+      SOURCE c-index-test.c
+      APPEND_STRING PROPERTY COMPILE_FLAGS " -Wno-c11-extensions"
+      )
+  endif()
 endif()
 
 if (LLVM_BUILD_STATIC)

